### ISSUE TITLE
Almost all tests fail, result in "but was:<[]>"

### ISSUE THREAD

**Issue by gregbown** [angular/clutz#363]
Doing the initial build; windows 10, typescript@next, gradle 3.1, Java 1.8.092 I got a build failure 125 tests, 111 failures. Any suggestions? I seem to be getting all "but was:<[]>" in the results. Thank you.

Running "gradle build installDist" in cmd
I assume there is a step missing in the build documentation?

Thank you
Greg

**Comment by gmoothart** [comment]
I don't have a Windows machine to test with :-(

Probably it will work if you run it in a linux vm.

**Comment by mprobst** [comment]
This sounds as if you don't have clang-format installed, or source
formatting fails for some reason? I think our test runner currently doesn't
properly check for a clang-format failure :-(

>

**Comment by gregbown** [comment]
You are correct, I did not have clang installed. The only resource I found for clang on windows was http://llvm.org/builds/
Installing this did not change the outcome although I'm sure there is more to the configuration for windows. Any further suggestions would be welcome,
Thank you

**Comment by dpurp** [comment]
@gregbown: Can you verify that clang-format is on your system PATH?

If you are using Eclipse, you can also extend your PATH in the "Environment" section  of "Run configurations."

See: http://stackoverflow.com/a/12810433

**Comment by mprobst** [comment]
I think easiest way to install is to run "npm install clang-format". You'll
need nodejs for that. While at it, you'll also need to install
typescript@beta from npm.

David Purpura <notifications@github.com> schrieb am Mo., 3. Okt. 2016,
13:56:

**Comment by gregbown** [comment]
I have installed clang from npm, also, the clutz docs recomend typescript@next which is what I have?
The gradle build clearly references node_modules however npm install is never mentioned in the documentation and there are no dependencies listed in package.json. Again, it seems there are crucial details missing from the documentation and project.

My results are slightly different now, however, still getting mostly errors, correction, even more errors.
128 tests completed, 114 failed
:test FAILED

There is no difference in the results between typescript@beta and typescript@next

**Comment by mprobst** [comment]
@gregbown I've added some docs and deps in `package.json`. Can you confirm if this works for you, and if not, please reopen this issue? Thanks!

**Comment by gregbown** [comment]
Here are the steps I took to get this to work on:
Windows 10 Professional
Gradle 4.8
Typescript 2.9
google-java-format 1.6
clang-format 1.2.3
Cygwin

Downloaded the TypeScript SDK

**build.gradle**

changed path to clang from 
```node_modules/.bin/clang-format```
to
```node_modules/clang-format/bin/win32/clang-format.exe```

changed googleJavaFormat toolVersion to 1.6

**com/google/javascript/gents/TypeScriptGenerator.java**

changed path to clang from ```./node_modules/.bin/clang-format``` to ```./node_modules/clang-format/bin/win32/clang-format.exe```

**clutz/src/test/java/com/google/javascript/clutz/DeclarationSyntaxTest.java**

Changed path to tsc to point to tsc.exe from TypeScript SDK.
Actual path on Windows is C:\Program Files (x86)\Microsoft SDKs\TypeScript\2.9\tsc.exe
```
  public static final Path TSC =
      FileSystems.getDefault()
          .getPath("C:", "Program Files (x86)", "Microsoft SDKs", "TypeScript", "2.9", "tsc");
```

**clutz/src/test/java/com/google/javascript/gents/TypeScriptGeneratorMultiTests.java**

The following tests did not work on my environment because one or more of the inputs were an empty object.
```
// assertThat(transpiledSource).hasSize(sourceNames.size());

// assertThat(transpiledSource).containsKey(basename);
// assertThat(transpiledSource.get(basename)).isEqualTo(goldenText);
// assertThat(emittedLogItems).isEqualTo(goldenLogItems);
```

In addition to this I had to reformat about 12 Java classes with the google-java-format tool.  Initially Gradle complained that two classes were not formatted so I tried to format them with the tool but then realized build.gradle was referencing an out of date tool.  After I aligned the build with the latest tool Gradle stated 12 classes failed the format test.

One other super annoying thing I had to do was find a way to format all the d.ts, .ts and .js files to unix format because all the tests would fail due to extra carage returns or newlines.  To do this I added the dos2unix tool to cygwin and ran the tool on every test file.
